app.go

```diff
@@ -399,6 +400,36 @@ func (ehl errorHandlerList) HandleError(err error) {
 	}
 }
 
+// validate sets *App into validation mode without running invoked functions.
+func validate(validate bool) Option {
+	return &validateOption{
+		validate: validate,
+	}
+}
+
+type validateOption struct {
+	validate bool
+}
+
+func (o validateOption) apply(app *App) {
+	app.validate = o.validate
+}
+
+func (o validateOption) String() string {
+	return fmt.Sprintf("fx.validate(%v)", o.validate)
+}
+
+// ValidateApp validates that supplied graph would run and is not missing any dependencies. This
+// method does not invoke actual input functions.
+func ValidateApp(opts ...Option) error {
```

Comment:
I think originally this was proposed for the `fxtest` package. Is there a usecase for ValidateApp in the main package that we haven't discussed?

Reply:
Having `ValidateApp` in `fxtest` would require exposure of a `dry` getter/setter in Fx which isn't desired right now. We want a `dry` flag for us, not users (at least for this feature). So, in order to keep `dry` private, I moved the `ValidateApp` into main package. It also only returns an error anyway, so no need for any extra testing facilities. 

Advantage of exposing `Dry` to users would have been an easier migration for existing fxtests, where u would just need to add the extra option to existing tests and be able to remove all of the setup of side effects that you have. However, this is a minor advantage outweighed by exposure of `Dry` which we can avoid.

Reply:
I was thinking about how to get it done using an internal package, but the only solution I came up with is refactoring the options with an extra level of indirection. The `Option` interface would change its `apply(*App)` to an `apply(*internal.AppParams)`, and after applying all options we'd convert an `internal.AppParam` to `App`.

Agree it seems a little much, having it on fx is probably fine.